### PR TITLE
Closes #8724: onPageStart/Stop ignore initial loads of about:blank

### LINKED ISSUE #8724: onPageStart/onPageStop should ignore initial loads of about:blank
Back in https://github.com/mozilla-mobile/android-components/issues/403 we started ignoring initial loads of `about:blank` in `onLocationChange` and `onSecurityChange`, but we didn't handle it in `onPageStart` and `onPageStop` which means we're forwarding progress and loading state changes unnecessarily.

This can cause observers to run multiple times and it was in fact one cause of https://github.com/mozilla-mobile/android-components/issues/4795. The original problem here doesn't exist anymore (duplicate observer calls when a tab is opened and a URL loaded), but it is still a problem when loading a tab via an intent.

Since we're not forwarding initial `about:blank` events in `onLocationChange` we should also not foward for `onPageStart/onPageStop`.

### PR DESCRIPTION
**Pull request by csadilek**
See #8724: This is the same workaround we already applied for `onLocationChange` and `onSecurityChange`. We forgot to apply for `onPageStart`/`onPageStop` meaning we get our observers invoked unnecessarily and only partially e.g. we get loading state changes, but never a url change notified. 

This can cause unneeded UI rendering and incorrect telemetry. On the incorrect telemetry part: This happens because it's possible to see two loading state changes for the same session, one for about:blank (loading=true, then loading=false), the second one for the actual session url (also loading=true, then loading=false). The session and observers "don't know" the first ones are for about:blank which we're skipping for earlier and later about:blank events. So, this is mostly keeping things consistent :). But the workaround in general is still 🙈 .

### CODE REVIEW

File: components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt [hunk 1]
```diff
@@ -612,6 +621,12 @@ class GeckoEngineSession(
             // by the time we reach here, any new request will come from web content.
             // If it comes from the chrome, loadUrl(url) or loadData(string) will set it to
             // false.
+
+            // Ignore initial load of about:blank (see https://github.com/mozilla-mobile/android-components/issues/403)
+            if (initialLoad && pageLoadingUrl == ABOUT_BLANK) {
```
Comment (grigoryk) [inline components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt:626]: onPageStart and onPageStop will always be called on the same thread?

**Review by grigoryk — APPROVED** [review]
It'll be very nice if GV didn't send us garbage events!

File: components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt (hunk 1 above)
Comment (csadilek) [reply components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt:626]: Yeah, these are all invoked on Main.

**Comment by csadilek** [comment]
Thanks for the review! Did a round of smoke testing in Fenix and since https://github.com/mozilla-mobile/fenix/pull/15940 landed (sooner than planned) which removed a workaround for this, I will land this as well to make sure telemetry doesn't change.

**Pull request merged**